frontend/taipy-gui/src/components/Taipy/TimeSelector.tsx

```diff
@@ -0,0 +1,111 @@
+import { isValid } from "date-fns";
+import { MobileTimePicker } from '@mui/x-date-pickers/MobileTimePicker';
+import { getSuffixedClassNames, TaipyActiveProps, TaipyChangeProps, getCssSize } from "./utils";
+import { createSendUpdateAction } from "../../context/taipyReducers";
+import { useClassNames, useDispatch, useDynamicProperty, useModule } from "../../utils/hooks";
+import ErrorFallback from "../../utils/ErrorBoundary";
+import Field from "./Field";
+
+
+interface TimeSelectorProps extends TaipyActiveProps, TaipyChangeProps {
+  asClock?: boolean;
+  format?: string;
+  defaultDate?: string;
+  defaultTime?: string;
+  defaultEditable?: boolean;
+  editable?: boolean;
+  label?: string;
+  time: string;
+  width?: string | number;
+}
+
+const boxSx = { display: "inline-block" };
+const TimeSelector = (props: TimeSelectorProps) => {
+  const { asClock = false, id, updateVarName, propagate = true } = props;
+  const [value, setValue] = useState <Date | null> (null);
+
+  const dispatch = useDispatch();
+  const module = useModule();
+  const className = useClassNames(props.libClassName, props.dynamicClassName, props.className);
+  const active = useDynamicProperty(props.active, props.defaultActive, true);
+  const editable = useDynamicProperty(props.editable, props.defaultEditable, true);
+  const hover = useDynamicProperty(props.hoverText, props.defaultHoverText, undefined);
+  const timeSx = useMemo(() => (props.width ? { maxWidth: getCssSize(props.width) } : undefined), [props.width]);
+
+  const handleChange = useCallback(
+    (v: Date | null) => {
+        setValue(v);
+        if (v !== null && isValid(v)) {
+            dispatch(
+              createSendUpdateAction(
+                  updateVarName,
+                  v,
+                  module,
+                  props.onChange,
+                  propagate
+              )
+            );
+        }
+    },
+    [updateVarName, dispatch, propagate, props.onChange, module]
+  );
+
+  return (
+    <ErrorBoundary FallbackComponent={ErrorFallback}>
+      <Tooltip title={hover || ""}>
+        <Box id={id} className={`${className}`} sx={boxSx}>
+          {editable ? (
+            asClock ? (
+              <MobileTimePicker
+                value={value}
+                onChange={handleChange}
+                className={getSuffixedClassNames(className, "-picker")}
+                disabled={!active}
+                label={props.label}
+                format={props.format}
+                sx={timeSx}
+              />
+            ):(
+              <TimePicker 
+                value={value}
+                onChange={handleChange}
+                className={getSuffixedClassNames(className, "-picker")}
+                disabled={!active}
+                label={props.label}
+                format={props.format}
+                sx={timeSx}
+              />
+            )
+          ):(
+            <Field 
+              dataType="string"
```

Comment:
should we support a time dataType in Field ?

Reply:
we should, done